### PR TITLE
Extend ext/random documentation

### PR DESCRIPTION
**Pull request by TimWolla**

### CODE REVIEW

**Review by zeriyoshi — APPROVED** [review]

File: reference/random/functions/random-int.xml [hunk 1]
```diff
@@ -30,17 +30,17 @@
     <term><parameter>min</parameter></term>
     <listitem>
      <para>
-      The lowest value to be returned, which must be
-      <constant>PHP_INT_MIN</constant> or higher.
+      The lowest value to be returned. must be
+      <constant>PHP_INT_MIN</constant> or greater.
```
Comment (cmb69) [inline reference/random/functions/random-int.xml:34]: Either use a semicolon, or start the sentence with an upper case letter (should probably be a full sentence then). I'd prefer the former:
```suggestion
      The lowest value to be returned; must be
      <constant>PHP_INT_MIN</constant> or greater.
```

File: reference/random/functions/random-int.xml [hunk 2]
```diff
@@ -30,17 +30,17 @@
...
      </para>
     </listitem>
    </varlistentry>
    <varlistentry>
     <term><parameter>max</parameter></term>
     <listitem>
      <para>
-      The highest value to be returned, which must be less than or equal to
-      <constant>PHP_INT_MAX</constant>.
+      The highest value to be returned. Must be
```
Comment (cmb69) [inline reference/random/functions/random-int.xml:42]:
```suggestion
      The highest value to be returned; must be
```

File: reference/random/functions/random-int.xml [hunk 3]
```diff
@@ -50,8 +50,10 @@
  <refsect1 role="returnvalues"><!-- {{{ -->
   &reftitle.returnvalues;
   <para>
-   Returns a cryptographically secure random integer in the range
-   <parameter>min</parameter> to <parameter>max</parameter>, inclusive.
+   Returns an cryptographically securely uniformly selected integer from the closed interval
```
Comment (cmb69) [inline reference/random/functions/random-int.xml:53]: This looks gramatically wrong. I think it should be:
```suggestion
   Returns a cryptographically secure, uniformly selected integer from the closed interval
```

File: language-snippets.ent [hunk 4]
```diff
@@ -3766,6 +3766,15 @@ local: {
  </note>
 '>
 
+<!-- Random snippets -->
+<!ENTITY random.engineErrors '
+ <listitem xmlns="http://docbook.org/ns/docbook">
+  <simpara>
+   Any <classname>Throwable</classname>s thrown by the <methodname>Random\Engine::generate</methodname> method
+   of the underlying <varname linkend="random-randomizer.props.engine">Random\Randomizer::$engine</varname>.
```
Comment (cmb69) [inline language-snippets.ent:3774]: Hmm, `<varname>` can't be linked this way; the element needs to be `<link>` instead. Wrapping a `<link>` around `<varname>` doesn't render the link, though, so we should probably use (at least for now):
```suggestion
   of the underlying <link linkend="random-randomizer.props.engine"><literal>Random\Randomizer::$engine</literal></link>.
```

**Review by cmb69 — COMMENTED** [review]
Thank you!

File: language-snippets.ent [hunk 5]
```diff
@@ -3737,7 +3737,7 @@ local: {
  <listitem>
   <simpara>
    If none of the aforementioned sources are available, then a
-   <classname>\Random\RandomException</classname> will be thrown.
+   <classname>Random\RandomException</classname> will be thrown.
```
[the remark this answers is not in the document]
Comment (TimWolla) [reply language-snippets.ent:3740]: To clarify: I prefer the FQN variant, but linking to the class only works without the leading backslash, so there's effectively only one option anyway.

File: reference/random/random/randomizer/getint.xml [hunk 6]
```diff
@@ -27,15 +27,17 @@
     <term><parameter>min</parameter></term>
     <listitem>
      <para>
-
+      The lowest value to be returned. must be
+      <constant>PHP_INT_MIN</constant> or greater.
```
Comment (TimWolla) [inline reference/random/random/randomizer/getint.xml:31]:
```suggestion
      The lowest value to be returned; must be
      <constant>PHP_INT_MIN</constant> or greater.
```

File: reference/random/random/randomizer/getint.xml [hunk 7]
```diff
@@ -27,15 +27,17 @@
...
      </para>
     </listitem>
    </varlistentry>
    <varlistentry>
     <term><parameter>max</parameter></term>
     <listitem>
      <para>
-
+      The highest value to be returned. Must be
+      <constant>PHP_INT_MAX</constant> or less.
```
Comment (TimWolla) [inline reference/random/random/randomizer/getint.xml:40]:
```suggestion
      The highest value to be returned; must be
      <constant>PHP_INT_MAX</constant> or less.
```

File: reference/random/random/randomizer/construct.xml [hunk 8]
```diff
@@ -26,7 +26,11 @@
     <term><parameter>engine</parameter></term>
     <listitem>
      <para>
+      The <classname>Random\Engine</classname> to use to generate randomness.
+     </para>
 
+     <para>
+      If <parameter>engine</parameter> is omitted or null, a new <classname>Random\Engine\Secure</classname> object will be used.
```
Comment (cmb69) [inline reference/random/random/randomizer/construct.xml:33]: Missed that in the first round:
```suggestion
      If <parameter>engine</parameter> is omitted or &null;, a new <classname>Random\Engine\Secure</classname> object will be used.
```

**Review by cmb69 — APPROVED** [review]
A minor issue remains to be fixed, but otherwise this is good to be merged.

**Comment by TimWolla** [comment]
@cmb69 Also "Squash and Merge" for the docs?

**Comment by cmb69** [comment]
> Also "Squash and Merge" for the docs?

Yes, please.

**Pull request merged**